contracts/crowdfund/AtomicManualParty.sol

```diff
@@ -0,0 +1,128 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity 0.8.20;
+
+import { IGlobals } from "../globals/IGlobals.sol";
+import { LibGlobals } from "../globals/LibGlobals.sol";
+import { IPartyFactory } from "../party/IPartyFactory.sol";
+import { Party } from "../party/Party.sol";
+import { IERC721 } from "../tokens/IERC721.sol";
+import { MetadataProvider } from "../renderers/MetadataProvider.sol";
+import { Proxy } from "../utils/Proxy.sol";
+import { Implementation } from "../utils/Implementation.sol";
+
+/// @title AtomicManualParty
+/// @notice Singleton that is called to create a party manually with an array
+///         of party members and their votes.
+contract AtomicManualParty {
+    /// @notice Returned if the `AtomicManualParty` is created with no members
+    error NoPartyMembers();
+    /// @notice Returned if the lengths of `partyMembers` and `partyMemberVotes` do not match
+    error PartyMembersArityMismatch();
+
+    // The `Globals` contract storing global configuration values. This contract
+    // is immutable and it’s address will never change.
+    IGlobals private immutable _GLOBALS;
+
+    constructor(IGlobals globals) {
+        _GLOBALS = globals;
+    }
+
+    /// @notice Atomically creates the party and distributes the party cards
+    function createParty(
+        Party partyImpl,
+        Party.PartyOptions memory opts,
+        IERC721[] memory preciousTokens,
+        uint256[] memory preciousTokenIds,
+        uint40 rageQuitTimestamp,
+        address[] memory partyMembers,
+        uint96[] memory partyMemberVotes
```

Comment:
Do we need to check for zeros in this array? What are the consequences if some members are assigned zero voting power?

Reply:
Not 100% sure but clearly the intended use-case is non-zero power and non-zero address. Might be worth adding just for the sanity check anyways.

Reply:
Yeah, I think it's worth the check here.